lib/src/scoring.dart

```diff
@@ -0,0 +1,192 @@
+  }
+
+  final avgY = 0.75;
+  final t = solveCubicSpecial(-3 * normAvg, 3 * normAvg, -1 * normValue);
+
+  var t2 = math.pow(t, 3) - (3 * avgY * math.pow(t, 2)) + (3 * t * avgY);
+
+  if (t2 <= epsilon) {
+    return 0;
+  }
+
+  if (t2 >= (1 - epsilon)) {
+    return 1;
+  }
+
+  return t2;
+}
+
+/// Special case of [solveCubic] which supports only code paths needed by
+/// [calculateBezierScore].
+///
+/// Inspired by https://stackoverflow.com/a/27176424/39827
+num solveCubicSpecial(num b, num c, num d) {
+  // Convert to depressed cubic t^3+pt+q = 0 (subst x = t - b/3a)
+  var p = (3 * c - b * b) / 3;
+  var q = (2 * b * b * b - 9 * b * c + 27 * d) / 27;
+
+  num root;
+
+  assert(abs(p) >= epsilon);
+
+  if (abs(q) < epsilon) {
+    assert(p >= 0);
+    root = 0;
+  } else {
+    var D = q * q / 4 + p * p * p / 27;
+    assert(abs(D) >= epsilon);
+    assert(D > 0);
+    var u = cubeRoot(-q / 2 - math.sqrt(D));
+    root = u - p / (3 * u);
+  }
+
+  root -= b / 3;
+
+  return root;
+}
+
+// Inspired by https://stackoverflow.com/a/27176424/39827
+// **Deprecated**
+// ("Kept here for completeness. Use `solveCubicSpecial` for scoring.")
+List<num> solveCubic(num a, num b, num c, num d) {
+  if (abs(a) < epsilon) {
+    // Quadratic case, ax^2+bx+c=0
+    a = b;
+    b = c;
+    c = d;
+    if (abs(a) < epsilon) {
+      // Linear case, ax+b=0
+      a = b;
+      b = c;
+      if (abs(a) < epsilon) // Degenerate case
+        return [];
+      return [-b / a];
+    }
+
+    var D = b * b - 4 * a * c;
+    if (abs(D) < epsilon)
+      return [-b / (2 * a)];
+    else if (D > 0)
+      return [(-b + math.sqrt(D)) / (2 * a), (-b - math.sqrt(D)) / (2 * a)];
+    return [];
+  }
+
+  // Convert to depressed cubic t^3+pt+q = 0 (subst x = t - b/3a)
+  var p = (3 * a * c - b * b) / (3 * a * a);
+  var q = (2 * b * b * b - 9 * a * b * c + 27 * a * a * d) / (27 * a * a * a);
+  var roots;
+
+  if (abs(p) < epsilon) {
+    // p = 0 -> t^3 = -q -> t = -q^1/3
+    roots = [cubeRoot(-q)];
+  } else if (abs(q) < epsilon) {
+    // q = 0 -> t^3 + pt = 0 -> t(t^2+p)=0
+    roots = <num>[0]..addAll(p < 0 ? [math.sqrt(-p), -math.sqrt(-p)] : []);
+  } else {
+    var D = q * q / 4 + p * p * p / 27;
+    if (abs(D) < epsilon) {
+      // D = 0 -> two roots
+      roots = [-1.5 * q / p, 3 * q / p];
+    } else if (D > 0) {
+      // Only one real root
+      var u = cubeRoot(-q / 2 - math.sqrt(D));
+      roots = [u - p / (3 * u)];
+    } else {
+      // D < 0, three roots, but needs to use complex numbers/trigonometric solution
+      var u = 2 * math.sqrt(-p / 3);
+      var t = math.acos(3 * q / p / u) /
+          3; // D < 0 implies p < 0 and acos argument in [-1..1]
+      var k = 2 * math.PI / 3;
+      roots = [u * math.cos(t), u * math.cos(t - k), u * math.cos(t - 2 * k)];
+    }
+  }
+
+  // Convert back from depressed cubic
+  for (var i = 0; i < roots.length; i++) roots[i] -= b / (3 * a);
+
+  return roots;
+}
+
+num abs(a) => a < 0 ? -a : a;
```

Comment:
why not a.abs() ?
https://api.dartlang.org/stable/1.23.0/dart-core/num/abs.html